contracts/BIFI/strategies/Stellaswap/StrategyStellaswapDualNonNativeLP.sol

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+import "../../interfaces/common/IUniswapRouter.sol";
+import "../../interfaces/common/IUniswapV2Pair.sol";
+import "../../interfaces/traderjoe/IMasterChef.sol";
+import "../Common/StratManager.sol";
+import "../Common/FeeManager.sol";
+
+/**
+ * Used to compound StellaDistributorV2
+ */
+contract StrategyStellaswapDualNonNativeLP is StratManager, FeeManager {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    // Tokens used
+    address public native;
+    address public output;
+    address public secondOutput;
+    address public want;
+    address public lpToken0;
+    address public lpToken1;
+
+    // Third party contracts
+    address public chef;
+    uint256 public poolId;
+
+    bool public harvestOnDeposit;
+    uint256 public lastHarvest;
+
+    // Routes
+    address[] public outputToNativeRoute;
+    address[] public secondOutputToNativeRoute;
+    address[] public nativeToLp0Route;
+    address[] public nativeToLp1Route;
+
+    /**
+     * @dev Event that is fired each time someone harvests the strat.
+     */
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+
+    constructor(
+        address _want,
+        uint256 _poolId,
+        address _chef,
+        address _vault,
+        address _unirouter,
+        address _keeper,
+        address _strategist,
+        address _beefyFeeRecipient,
+        address[] memory _outputToNativeRoute,
+        address[] memory _secondOutputToNativeRoute,
+        address[] memory _nativeToLp0Route,
+        address[] memory _nativeToLp1Route
+    ) StratManager(_keeper, _strategist, _unirouter, _vault, _beefyFeeRecipient) public {
+        want = _want;
+        poolId = _poolId;
+        chef = _chef;
+
+        output = _outputToNativeRoute[0];
+        native = _outputToNativeRoute[_outputToNativeRoute.length - 1];
+        outputToNativeRoute = _outputToNativeRoute;
+
+        secondOutput = _secondOutputToNativeRoute[0];
+        secondOutputToNativeRoute = _secondOutputToNativeRoute;
```

Comment:
Could implement route checking to ensure that `secondOutputToNativeRoute == native`, similar to the reuqire statements below

Reply:
fixed